include/swift/ClangImporter/CXXMethodBridging.h

```diff
@@ -8,70 +8,73 @@
 #include <string>
 namespace swift {
 struct CXXMethodBridging {
-  enum class Kind { unkown, getter, setter, subscript };
+  enum class Kind { unknown, getter, setter, subscript };
 
-  enum class NameKind { unkown, snake, lower, camel, title };
+  enum class NameKind { unknown, snake, lower, camel, title };
 
   CXXMethodBridging(const clang::CXXMethodDecl *method) : method(method) {}
 
   Kind classify() {
     if (nameIsBlacklist())
-      return Kind::unkown;
+      return Kind::unknown;
 
     // this should be handled as snake case. See: rdar://89453010
     // case. In the future we could
     //  import these too, though.
     auto nameKind = classifyNameKind();
     if (nameKind != NameKind::title && nameKind != NameKind::camel &&
-        nameKind != NameKind::lower)
-      return Kind::unkown;
+        nameKind != NameKind::lower && nameKind != NameKind::snake)
+      return Kind::unknown;
 
     if (getClangName().startswith_insensitive("set")) {
       // Setters only have one parameter.
       if (method->getNumParams() != 1)
-        return Kind::unkown;
+        return Kind::unknown;
 
       // rdar://89453106 (We need to handle imported properties that return a
       // reference)
       if (method->getParamDecl(0)->getType()->isReferenceType())
-        return Kind::unkown;
+        return Kind::unknown;
 
       return Kind::setter;
     }
 
     // Getters and subscripts cannot return void.
     if (method->getReturnType()->isVoidType())
-      return Kind::unkown;
+      return Kind::unknown;
 
     if (getClangName().startswith_insensitive("get")) {
       // Getters cannot take arguments.
       if (method->getNumParams() != 0)
-        return Kind::unkown;
+        return Kind::unknown;
 
       // rdar://89453106 (We need to handle imported properties that return a
       // reference)
       if (method->getReturnType()->isReferenceType())
-        return Kind::unkown;
+        return Kind::unknown;
 
       return Kind::getter;
     }
 
     // rdar://89453187 (Add subscripts clarification to CXXMethod Bridging to
     // clean up importDecl)
-    return Kind::unkown;
+    return Kind::unknown;
   }
 
   NameKind classifyNameKind() {
-    bool allLower = llvm::all_of(getClangName(), islower);
+    bool hasUpper = llvm::any_of(getClangName(), std::isupper);
 
-    if (getClangName().empty())
-      return NameKind::unkown;
-
-    if (getClangName().contains('_'))
-      return allLower ? NameKind::snake : NameKind::unkown;
+    // if the string is empty
+    if (getClangName().empty()) {
+      return NameKind::unknown;
+    }
 
-    if (allLower)
+    // if one of the elements is an underscore
```

Comment:
In the future, it doesn't make a lot of sense to comment when the line directly below says the same thing as the comment. Most programmers can fairly easily derive "if one of the elements is an underscore" from "if contains '_'."

Comments are more helpful when they describe _why_ we care about this case. For example "an underscore in the name means it's snake case."

Anyway, no need to change anything here. Just for the future.

Reply:
Oh yeah that makes sense, thank you!